File: PRACTICA PRIMER PARCIAL/Ejercicio6/Python/NumeroPrimo.py
def es_primo(numero):
    if numero < 2:
        return False
    
    for i in range(2, numero):
        if numero % i == 0: # Si el resto de la división es 0, no es primo.
            return False
    
    return True

numero = int(input("Ingrese un número de un solo dígito: "))

if es_primo(numero):
    print("El número es primo.")
else:
    print("El número no es primo.")

File: PRACTICA PRIMER PARCIAL/Ejercicio6/Java/NumeroPrimo.java
import java.util.Scanner;

public class NumeroPrimo {
    public static boolean esPrimo(int numero) {
        if (numero < 2) {
            return false;
        }
        
        for (int i = 2; i < numero; i++) {
            if (numero % i == 0) {
                return false;
            }
        }
        
        return true;
    }
    
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        System.out.print("Ingrese un número de un solo dígito: ");
        int numero = scanner.nextInt();
        
        if (esPrimo(numero)) {
            System.out.println("El número es primo.");
        } else {
            System.out.println("El número no es primo.");
        }
        
        scanner.close();
    }
}
